plugin/keyring_vault/vault_curl.h

```diff
@@ -18,17 +18,16 @@ namespace keyring
 class Vault_curl : public IVault_curl, private boost::noncopyable
 {
 public:
-  Vault_curl(ILogger *logger)
+  Vault_curl(ILogger *logger, CURL *curl)
```

Comment:
This is due to cento6 failure. The problem with CURL in centos6 is that it fails when another instance of CURL object is created and initialized. The default CURL in centos6 is using NSS library as the SSL layer. Somehow, when I initialize another CURL "session" NSS library correctly discovers that it was already initialized but fails to find ssl layer descriptor. This is what I gathered from debugging - debug versions of this library. The same problem was when I was trying to reuse exisiting CURL object - first I had to call curl_easy_reset and I had the same problem. The workaround which works is to completely reinitialize whole CURL together with SSL session (by CURL global). This operation will be only done when keyring_vault_config variable gets changed - the CURL initialization is done in the highest layer - vault_keyring.cc and I thought it will be better if I keep the initialization related things in this layer. As you see - CURL is working in this setup and it was not easy to make it work - curl on centos6 is 9 years old. Thus I would really like to leave it as it is - this works for all unit tests/mtr tests.